src/dfcx_scrapi/tools_ml/utterance_generator_utils.py

```diff
@@ -0,0 +1,369 @@
+        
+        Creates a test dataset where none of the utterances in the test dataset
+        are in the training of the existing phrases.
+
+        Args: 
+            agent_id: ID of the DFCX agent.
+            intent_subset: intents to generate a test dataset for.
+            dataset_size: number of synthetic phrases to generate, default 100.
+        Returns:
+            a DataFrame containing synthetic test dataset utterances, columns:
+                id: IDs of original utterances.
+                synthetic instances: number of synthetic phrases per original 
+                    phrase. 
+                utterance: original utterances.
+                synthetic_phrases: generated phrases.
+                intent: intent the utterance is from.
+        """
+        training_phrases = self.intents.bulk_intent_to_df(
+            agent_id=agent_id, intent_subset=intent_subset
+        )
+        training_phrases = training_phrases.copy().rename(
+            columns={"tp": "utterance"})
+
+        test_dataset = self._generate_phrases(training_phrases, dataset_size)
+        test_dataset = test_dataset[:dataset_size]
+        return test_dataset.reset_index(drop=True)
+
+    def create_test_dataset(
+        self, 
+        agent_id: str, 
+        intent_subset: List[str], 
+        dataset_size: int = 100
+    ) -> pd.DataFrame:
+        """Creates a test dataset for a given list of intents. 
+        
+        The phrases in this dataset will not be exact string match phrases which
+        exist in the training phrases but will be close semantically. This set
+        is automatically labeled by the intent whose training was used to
+        generate the new phrase. This can be used to run through the
+        core.conversations run_intent_detection function. You may need to
+        specify a flow_display_name and page_display_name in the dataframe to
+        run the set at the correct location.
+
+        Args:
+            agent_id: name parameter of the agent to pull intents from
+                full path to agent
+            intent_subset: display names of the intents to create a test
+                for, base phrases come from the training in the intent.
+            dataset_size: overall target size of the test set to create, may
+                be less depending if new independent phrases can be generated
+                from the data. The function tries to get even entries per
+                intent.
+
+        Returns:
+            Dataframe with columns:
+                utterance: synthesized phrases.
+                intent: intent the utterance was generated from, 
+                    also true label.
+        """
+        synthetic_dataset = self.create_synthetic_dataset(agent_id, intent_subset, 
+            dataset_size)
+        test_dataset = (
+            synthetic_dataset.copy()[["synthetic_phrases", "intent"]]
+            .rename(columns={"synthetic_phrases": "utterance"})
```

Comment:
@Hgithubacct Can we keep the intent column's name as "intent" and remove the rename to "display_name"? Just for consistent namespace.